subprojects/plugin-development/src/integTest/groovy/org/gradle/plugin/devel/tasks/ValidateTaskPropertiesIntegrationTest.groovy

```diff
@@ -387,6 +391,58 @@ class ValidateTaskPropertiesIntegrationTest extends AbstractIntegrationSpec {
         file("build/reports/task-properties/report.txt").text == ""
     }
 
+    @Unroll
+    def "report setters for mutable type #type"() {
+        @Language("JAVA") myTask = """
+            import org.gradle.api.DefaultTask;
+            import org.gradle.api.tasks.InputFiles;
+
+            public class MyTask extends DefaultTask {
+                private ${type} sources;
+                
+                @InputFiles
+                public ${type} getSources() { return sources; } 
+                public void setSources(${type} sources) { this.sources = sources; } 
+            }
+        """
+        file("src/main/java/MyTask.java") << myTask
+
+        when:
+        fails "validateTaskProperties"
+
+        then:
+        file("build/reports/task-properties/report.txt").text == """
+            Warning: Type 'MyTask': 'sources' of type '${type.replaceAll("<.+>", "")}' has redundant setter method.
+        """.stripIndent().trim()
+
+        where:
+        type << [ConfigurableFileCollection.name, "${Property.name}<String>", RegularFileProperty.name]
+    }
+
+    @Unroll
+    def "don't report when non-final field doesn't have setters for type #type"() {
+        @Language("JAVA") myTask = """
+            import org.gradle.api.DefaultTask;
+            import org.gradle.api.tasks.InputFiles;
+
+            public class MyTask extends DefaultTask {
+                private final ${type} sources;
+                
+                MyTask() { sources = null; }
+                
+                @InputFiles
+                public ${type} getSources() { return sources; } 
+            }
+        """
+        file("src/main/java/MyTask.java") << myTask
+
+        expect:
+        succeeds "validateTaskProperties"
+
+        where:
+        type << [ConfigurableFileCollection.name, "${Property.name}<String>", RegularFileProperty.name]
+    }
+
```

Comment:
We should have a test case to check what happens when there's a lone setter without a getter. (We should probably have a specific warning about write-only properties for that case. I'm fine with doing that as a separate PR if you want, too.)

Reply:
I don't mind putting it here, I'm just not sure what to do in the write-only case :/

Reply:
I've added the test for the setter -> it doesn't fail yet, as we're working based on getters.
I can add it in a new PR, since implementing that does seem like a different feature.

Reply:
Let's add a `// TODO` above the setter-only property then. Please also open an issue to warn about setter-only properties.

Reply:
Done, opened: https://github.com/gradle/gradle/issues/9341